app/src/main/java/nerd/tuxmobil/fahrplan/congress/schedule/MainActivity.kt

```diff
@@ -0,0 +1,493 @@
+            appRepository.loadSchedule(
+                url = url,
+                onFetchingDone = ::onGotResponse,
+                onParsingDone = ::onParseDone,
+                onLoadingShiftsDone = ::onLoadShiftsDone
+            )
+        } else {
+            Log.d(LOG_TAG, "Fetching schedule already in progress.")
+        }
+    }
+
+    override fun onDestroy() {
+        appRepository.cancelLoading()
+        hideProgressDialog()
+        super.onDestroy()
+    }
+
+    override fun onResume() {
+        super.onResume()
+        isScreenLocked = keyguardManager.isKeyguardLocked
+        val sidePaneView = findViewById<FragmentContainerView>(R.id.detail)
+        if (sidePaneView != null && isFavoritesInSidePane) {
+            sidePaneView.isVisible = !isScreenLocked
+        }
+        if (!appRepository.readScheduleChangesSeen()) {
+            showChangesDialog()
+        }
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu): Boolean {
+        super.onCreateOptionsMenu(menu)
+        menuInflater.inflate(R.menu.mainmenu, menu)
+        return true
+    }
+
+    private fun showChangesDialog() {
+        val fragment = findFragment(ChangesDialog.FRAGMENT_TAG)
+        if (fragment == null) {
+            val sessions = appRepository.loadChangedSessions()
+            val meta = appRepository.readMeta()
+            val statistic = ChangeStatistic.of(sessions)
+            ChangesDialog
+                .newInstance(meta.version, statistic)
+                .show(supportFragmentManager, ChangesDialog.FRAGMENT_TAG)
+        }
+    }
+
+    private fun showAboutDialog() {
+        val meta = appRepository.readMeta()
+        val transaction = supportFragmentManager.beginTransaction()
+        transaction.addToBackStack(null)
+        AboutDialog
+            .newInstance(meta.version, meta.subtitle, meta.title)
+            .show(transaction, AboutDialog.FRAGMENT_TAG)
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        when (item.itemId) {
+            R.id.menu_item_about -> {
+                showAboutDialog()
+                return true
+            }
+            R.id.menu_item_alarms -> {
+                AlarmList.startForResult(this)
+                return true
+            }
+            R.id.menu_item_settings -> {
+                SettingsActivity.startForResult(this)
+                return true
+            }
+            R.id.menu_item_schedule_changes -> {
+                openSessionChanges()
+                return true
+            }
+            R.id.menu_item_favorites -> {
+                openFavorites()
+                return true
+            }
+        }
+        return super.onOptionsItemSelected(item)
+    }
+
+    private fun openSessionDetails(sessionId: String) {
+        if (appRepository.updateSelectedSessionId(sessionId)) {
+            val sidePaneView = findViewById<FragmentContainerView>(R.id.detail)
+            if (sidePaneView == null) {
```

Comment:
`sidePaneView !is View` or reverse the order of the blocks and use `sidePaneView is View`